net/core/filter.c

```diff
@@ -5849,6 +5855,278 @@ static const struct bpf_func_proto bpf_tcp_gen_syncookie_proto = {
 
 #endif /* CONFIG_INET */
 
+#ifdef CONFIG_XDP_LUA
+
+static inline void verify_and_lock(void) {
+	struct xdplua_create_work *lw;
+
+	lw = this_cpu_ptr(&luaworks);
+	if (!lw->init) {
+		lw->init = true;
+		spin_lock(&lw->lock);
+	}
+}
+
+BPF_CALL_2(bpf_lua_dataref, struct xdp_buff *, ctx, int, offset) {
+	if (offset + ctx->data < ctx->data_end) {
+		int data_ref;
+
+		verify_and_lock();
+		data_ref = ldata_newref(ctx->L, ctx->data + offset,
+				ctx->data_end - ctx->data - offset);
+		return data_ref;
+	}
+
+	return -1;
+}
+
+static const struct bpf_func_proto bpf_lua_dataref_proto = {
+	.func		= bpf_lua_dataref,
+	.gpl_only	= false,
+	.pkt_access	= false,
+	.ret_type	= RET_INTEGER,
+	.arg1_type	= ARG_PTR_TO_CTX,
+	.arg1_type	= ARG_ANYTHING,
+};
+
+BPF_CALL_2(bpf_lua_dataunref, struct xdp_buff *, ctx, int, data_ref) {
+	verify_and_lock();
+	ldata_unref(ctx->L, data_ref);
+	return 0;
+}
+
+static const struct bpf_func_proto bpf_lua_dataunref_proto = {
+	.func		= bpf_lua_dataunref,
+	.gpl_only	= false,
+	.pkt_access	= false,
+	.ret_type	= RET_VOID,
+	.arg1_type	= ARG_PTR_TO_CTX,
+	.arg2_type	= ARG_ANYTHING,
+};
+
+BPF_CALL_4(bpf_lua_pcall, struct xdp_buff *, ctx, char *, funcname,
```

Comment:
@VictorNogueiraRio I believed we have already discussed that we should just wrap `lua_pcall` and allow the caller to handle the errors..